### PR TITLE
refactor(image_projection_based_fusion): add JSON Schema and remove default value spefications

### PR DESCRIPTION
**Pull request by ktro2828**
## Description

- Add ROS node configuration with JSON Schema as discussed in https://github.com/orgs/autowarefoundation/discussions/3371
- Remove default parameter specified in declare_parameter()

## Tests performed

Not applicable.

## Effects on system behavior

Not applicable.

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by ktro2828** [comment]
@yukke42 @tzhong518 @badai-nguyen Could check the parameters for the packages which each of you worked?

File: perception/image_projection_based_fusion/launch/roi_cluster_fusion.launch.xml [hunk 1]
```diff
@@ -18,23 +17,15 @@
...
   <arg name="sync_param_path" default="$(find-pkg-share image_projection_based_fusion)/config/roi_sync.param.yaml"/>
-  <arg name="remove_unknown" default="true"/>
-  <arg name="trust_distance" default="100.0"/>
 
   <!-- for eval variable-->
   <arg name="input_rois_number" default="$(var input/rois_number)"/>
 
   <!-- debug -->
   <!-- cspell: ignore minx, maxx, miny, maxy, minz, maxz  -->
   <arg name="debug_mode" default="false"/>
```
Comment (badai-nguyen) [inline perception/image_projection_based_fusion/launch/roi_cluster_fusion.launch.xml:28]: @ktro2828 shall we also move this to (roi_sync).param.yaml too?

**Comment by badai-nguyen** [comment]
@ktro2828 Thank you for your PR. Since the roi_pointcloud_fusion node was added recently, I really appreciate if you also add scheme json for that node's params.   https://github.com/autowarefoundation/autoware.universe/blob/c552c3dedc82f61015a918e505375d0bd04cde6c/perception/image_projection_based_fusion/src/roi_pointcloud_fusion/node.cpp#L35-L37

File: perception/image_projection_based_fusion/launch/roi_cluster_fusion.launch.xml (hunk 1 above)
Comment (ktro2828) [reply perception/image_projection_based_fusion/launch/roi_cluster_fusion.launch.xml:28]: I removed all `debug_mode` variables in launcher and moved into `roi_sync.params.yaml` in https://github.com/autowarefoundation/autoware.universe/pull/4902/commits/16faca87b7bec0b3be8ef8f18a6103d06ed97f0b

**Comment by ktro2828** [comment]
@badai-nguyen I add configuration and schema for `roi_pointcloud_fusion` in https://github.com/autowarefoundation/autoware.universe/pull/4902/commits/130791e7874ab66c37a1a7aa4e448c4b1394cf7c

File: perception/image_projection_based_fusion/config/pointpainting.param.yaml [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
+    model_params:
+      class_names: ["CAR", "TRUCK", "BUS", "BICYCLE", "PEDESTRIAN"]
+      paint_class_names: ["CAR", "BICYCLE", "PEDESTRIAN"]
+      point_feature_size: 7 # x, y, z, time-lag and car, pedestrian, bicycle
+      max_voxel_size: 40000
+      point_cloud_range: [-121.6, -76.8, -3.0, 121.6, 76.8, 5.0]
+      voxel_size: [0.32, 0.32, 8.0]
+      downsample_factor: 1
+      encoder_in_feature_size: 12
+      yaw_norm_thresholds: [0.3, 0.3, 0.3, 0.3, 0.0]
```
Comment (badai-nguyen) [inline perception/image_projection_based_fusion/config/pointpainting.param.yaml:12]:
```suggestion
      yaw_norm_thresholds: [0.3, 0.3, 0.3, 0.3, 0.0]
      has_twist: false
```

File: perception/image_projection_based_fusion/config/pointpainting.param.yaml [hunk 3]
```diff
@@ -1,18 +1,24 @@
...
+    densification_params:
+      world_frame_id: "map"
+      num_past_frames: 0
+    post_process_params:
+      # post-process params
+      circle_nms_dist_threshold: 0.3
+      iou_nms_target_class_names: ["CAR"]
+      iou_nms_search_distance_2d: 10.0
+      iou_nms_threshold: 0.1
```
Comment (badai-nguyen) [inline perception/image_projection_based_fusion/config/pointpainting.param.yaml:21]:
```suggestion
      iou_nms_threshold: 0.1
      score_threshold: 0.4
```

File: perception/image_projection_based_fusion/config/pointpainting.param.yaml [hunk 4]
```diff
@@ -1,18 +1,24 @@
...
+    omp_params:
+      # omp params
+      omp_num_threads: 1
```
Comment (badai-nguyen) [inline perception/image_projection_based_fusion/config/pointpainting.param.yaml:24]:
```suggestion
      num_threads: 1
```

Comment (badai-nguyen) [reply perception/image_projection_based_fusion/config/pointpainting.param.yaml:24]: These param `omp_params.num_threads`, `model_params.has_twist`, `post_process_params.score_threshold` also must be initialized?

File: perception/image_projection_based_fusion/launch/roi_pointcloud_fusion.launch.xml [hunk 5]
```diff
@@ -38,9 +36,7 @@
   <group>
     <node pkg="image_projection_based_fusion" exec="roi_pointcloud_fusion_node" name="roi_pointcloud_fusion" output="screen">
       <param name="rois_number" value="$(var input/rois_number)"/>
-      <param name="cluster_2d_tolerance" value="$(var cluster_2d_tolerance)"/>
-      <param name="min_cluster_size" value="$(var min_cluster_size)"/>
-      <param name="fuse_unknown_only" value="$(var fuse_unknown_only)"/>
+      <param from="(var param_path)"/>
```
Comment (badai-nguyen) [inline perception/image_projection_based_fusion/launch/roi_pointcloud_fusion.launch.xml:39]:
```suggestion
      <param from="$(var param_path)"/>
```
This is a typo?

**Comment by ktro2828** [comment]
@badai-nguyen Sorry for late to reaction, I fixed all you suggested.

**Comment by badai-nguyen** [comment]
@ktro2828 
All `image_projection_based_fusion` releaded nodes still cannot load with autoware 
`[ERROR] [roi_detected_object_fusion_node-51]: process has died [pid 57163, exit code -6, cmd '/home/autoware/autoware/install/image_projection_based_fusion/lib/image_projection_based_fusion/roi_detected_object_fusion_node --ros-args -r __node:=roi_detected_object_fusion -r __ns:=/perception/object_recognition/detection -p use_sim_time:=True -p wheel_radius:=0.383 -p wheel_width:=0.235 -p wheel_base:=2.79 -p wheel_tread:=1.64 -p front_overhang:=1.0 -p rear_overhang:=1.1 -p left_overhang:=0.128 -p right_overhang:=0.128 -p vehicle_height:=2.5 -p max_steer_angle:=0.7 --params-file /tmp/launch_params_y3act209 --params-file /home/autoware/autoware/install/image_projection_based_fusion/share/image_projection_based_fusion/config/roi_detected_object_fusion.param.yaml --params-file /home/autoware/autoware/install/autoware_launch/share/autoware_launch/config/perception/object_recognition/detection/image_projection_based_fusion/roi_sync.param.yaml --params-file /tmp/launch_params_1h8tfapn --params-file /tmp/launch_params_nl5xikak --params-file /tmp/launch_params_86kncd14 --params-file /tmp/launch_params_s7tl5q98 --params-file /tmp/launch_params_f7utsrw3 --params-file /tmp/launch_params_sj8c3jdo --params-file /tmp/launch_params_94a3_iuw --params-file /tmp/launch_params_70fyqug7 --params-file /tmp/launch_params_fc6kx_kj --params-file /tmp/launch_params_4s08xmb_ --params-file /tmp/launch_params_evx9gfnf --params-file /tmp/launch_params_bdmn5ypu --params-file /tmp/launch_params_0x0ky2ub --params-file /tmp/launch_params__v_2dkbe --params-file /tmp/launch_params_5ze5s4hs --params-file /tmp/launch_params_swedhxjf --params-file /tmp/launch_params_zcd9mfd2 --params-file /tmp/launch_params_mlp_7op8 --params-file /tmp/launch_params_9tp9l3yw --params-file /tmp/launch_params_w7wsi1c1 --params-file /tmp/launch_params_x8ora7is --params-file /tmp/launch_params__bgv8bwh --params-file /tmp/launch_params_4y8erxph --params-file /tmp/launch_params_ych09njm -r input:=centerpoint/objects -r output:=centerpoint/roi_fusion/objects'].
`

The reason seems related 
`[roi_detected_object_fusion_node-51]   what():  Statically typed parameter 'filter_scope_min_x' must be initialized.`

**Comment by badai-nguyen** [comment]
I think the [roi_sync.param.yaml](https://github.com/autowarefoundation/autoware_launch/blob/main/autoware_launch/config/perception/object_recognition/detection/image_projection_based_fusion/roi_sync.param.yaml) also need to be updated. I created PR for that https://github.com/autowarefoundation/autoware_launch/pull/739

**Review by badai-nguyen — APPROVED** [review]
LGTM

**Pull request merged**